pkg/server/route_prefix.go

```diff
@@ -0,0 +1,36 @@
+package server
+
+import (
+	"fmt"
+	"net/http"
+	"strings"
+)
+
+func WithRoutePrefix(mux *http.ServeMux, routePrefix string) *http.ServeMux {
+	// ensure that the route prefix begins with a slash
+	if !strings.HasPrefix(routePrefix, "/") {
+		routePrefix = "/" + routePrefix
+	}
+	// ensure route prefix doesn't have a trainling slash
+	routePrefix = strings.TrimSuffix(routePrefix, "/")
+
+	routePrefixMux := http.NewServeMux()
+	routePrefixMux.Handle(routePrefix+"/", http.StripPrefix(routePrefix, mux))
+	// Redirect to the route prefix if the user hits the root of the server
+	// e.g. if they port-forward
+	routePrefixMux.Handle("/", http.RedirectHandler(routePrefix+"/", http.StatusFound))
+
+	return routePrefixMux
+}
+
+func InjectHTMLBaseTag(html []byte, routePrefix string) []byte {
+	baseHref := routePrefix
+	// ensure baseHref begins and ends with a slash
+	if !strings.HasPrefix(baseHref, "/") {
+		baseHref = "/" + baseHref
+	}
```

Comment:
Should we calculate this once, and inject on every request?

Do we need to inject if if the RoutePrefix is "" ?

Reply:
Yeah, could do.

I don't think is very expensive to do on each request, but makes it clearer its not gonna change on each request etc.

> Do we need to inject if if the RoutePrefix is "" ?

Nope, the UI is designed to work without the `<base />` tag. We only add it if you're changing the base path etc. Will call that out in an improved comment here.

Reply:
> > Do we need to inject if if the RoutePrefix is "" ?
>
> Nope, the UI is designed to work without the <base /> tag. We only add it if you're changing the base path etc. Will call that out in an improved comment here.

I did not think this through and you are right, we do always need to inject it.